`vfscore`: Build error when using Clang

### Describe the bug

When using Clang to build a simple filesystem using program (no system call shim involved), the following error occurs:

```text
/home/razvan/orgs/unikraft/maintainer-tools/workdir/unikraft/lib/vfscore/main.c:1709:10: error: call to undeclared function 'uk_syscall_r_fstat'; ISO C99 and later do not support implicit function declarations [-Wi
mplicit-function-declaration]
 1709 |                 return uk_syscall_r_fstat((long) dirfd, (long) st);
```

### Steps to reproduce

Build [the `c-fs` application in the `catalog-core` repository](https://github.com/unikraft/catalog-core/tree/main/c-fs). Follow the steps:

1. Clone the repository:

   ```console
   git clone https://github.com/unikraft/catalog-core
   ```

1. Enter the `c-fs/` directory:

   ```console
   cd catalog-core/c-fs/
   ```
   
1. Switch to the `scripts` branch:

   ```console
   git checkout scripts
   ```

1. Build `c-fs` with Clang:

   ```console
   CC=/usr/bin/clang ./scr.ipts/build/qemu.x86_64
   ```

### Expected behavior

The build should be successful.

### Which architectures were you using or does this bug affect?

x86_64

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

```shell
/home/razvan/orgs/unikraft/maintainer-tools/workdir/unikraft/lib/vfscore/main.c:1709:10: error: call to undeclared function 'uk_syscall_r_fstat'; ISO C99 and later do not support implicit function declarations [-Wi
mplicit-function-declaration]
 1709 |                 return uk_syscall_r_fstat((long) dirfd, (long) st);
```

--- a/lib/vfscore/main.c
+++ b/lib/vfscore/main.c
@@ -1695,6 +1695,9 @@ UK_SYSCALL_R_DEFINE(int, lstat, const char*, pathname, struct stat*, st)
 	return __lxstat_helper(1, pathname, st);
 }
 
+/* The fstat syscall is no longer implemented here; need to declare */
+long uk_syscall_r_fstat(long dirfd, long st);
+
 static int __fxstatat_helper(int ver __unused, int dirfd, const char *pathname,
 		struct stat *st, int flags)
 {

